common/k8stest/util_apps.go

```diff
@@ -419,6 +509,21 @@ func (dfa *FioApp) Cleanup() error {
 				return fmt.Errorf("verify replicas before deleting PVC failed: %v", err)
 			}
 		}
+	} else {
+		err := DeleteDeployment(dfa.DeployName, common.NSDefault)
+		if err != nil {
+			return fmt.Errorf("failed to delete fio deployment %s, err: %v", dfa.DeployName, err)
+		}
+		const sleepTime = 3
+		 var pods *coreV1.PodList
+		for ix := 0; ix < (DefTimeoutSecs+sleepTime-1)/sleepTime; ix++ {
+			pods, err = GetDeploymentPods(dfa.DeployName, common.NSDefault)
+			logf.Log.Info("GetDeploymentPods", "error", err, "pods", pods)
+			if err != nil {
+				break
```

Comment:
why to exit loop in case of error ?

Reply:
We are expecting the deployment to be deleted/unavailable. If there is an error, it indicates that the deployment is unavailable

Reply:
Add extra check for not found like k8serrors.IsNotFound(err)

Reply:
you might need to remove error wrapping from `GetDeploymentPods` and `ListPodsWithLabel` to check for `IsNotFound`

Reply:
Fixed